File: app/src/main/java/com/github/alexeygorovoy/moxy_template/dagger/demo/splash/SplashComponent.java
package com.github.alexeygorovoy.moxy_template.dagger.demo.splash;

import com.github.alexeygorovoy.moxy_template.ui.demo.splash.view.SplashFragment;

import dagger.Subcomponent;


@Subcomponent(modules = {SplashModule.class})
public interface SplashComponent {
    void inject(SplashFragment fragment);
}

File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/dagger/demo/splash/SplashComponent.kt
package com.github.alexeygorovoy.moxytemplate.dagger.demo.splash

import com.github.alexeygorovoy.moxytemplate.ui.demo.splash.view.SplashFragment
import dagger.Component
import me.vponomarenko.injectionmanager.support.CompatInjectionManager

@Component(modules = [SplashModule::class], dependencies = [SplashDependencies::class])
interface SplashComponent {
    fun inject(fragment: SplashFragment)

    companion object {
        operator fun invoke(): SplashComponent = DaggerSplashComponent.builder()
            .splashDependencies(CompatInjectionManager.findComponent())
            .build()
    }
}
